cronsul

```diff
@@ -14,9 +14,22 @@ usage() {
 
 grab_lock_or_die() {
     local task_id="$1"
-    local schedule_date="$2"
+    local current_run="$2"
     local period="$3"
-    ret="$(curl -sS -X PUT -d "${HOSTNAME}" "${CONSUL_ADDR}/v1/kv/cronsul/${task_id}/${period}/${schedule_date}?cas=0")"
+    local query="${CONSUL_ADDR}/v1/kv/cronsul/${task_id}"
+
+    json="$(curl -sS -X GET "${query}")"
+    if [[ "$json" != "" ]]; then
+        local last_run="$(grep -Po '"Value":.*?[^\\]",' <<< ${json} | awk -F\" '{print $4}' | base64 --decode)"
```

Comment:
I'm pretty much fine with that, since I have jq installed most places, and it's available in package repos for a lot of distros. Would require a change to the README again.